In [3]:
%matplotlib inline
from sklearn import datasets 
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

In [7]:
digits = datasets.load_digits()
breast_cancer = datasets.load_breast_cancer()

In [17]:
X_d = digits.data
y_d = digits.target
X_b = breast_cancer.data
y_b = breast_cancer.target

In [28]:
from sklearn import cross_validation, naive_bayes
from sklearn.cross_validation import cross_val_score

clf_ber = naive_bayes.BernoulliNB()
clf_mult = naive_bayes.MultinomialNB()
clf_gaus = naive_bayes.GaussianNB()
clf_ber.fit(X_d,y_d)
clf_gaus.fit(X_d,y_d)
clf_mult.fit(X_d,y_d)
scores_d = [cross_val_score(clf_ber,X_d,y_d).mean(),
            cross_val_score(clf_gaus,X_d,y_d).mean(), 
            cross_val_score(clf_mult,X_d,y_d).mean()]
scores_d

[0.82582365077805819, 0.81860038035501381, 0.87087714897350532]

Максимальное - 0.87087714897350532, MultinomialNB

In [29]:
clf_ber = naive_bayes.BernoulliNB()
clf_mult = naive_bayes.MultinomialNB()
clf_gaus = naive_bayes.GaussianNB()
clf_ber.fit(X_b,y_b)
clf_gaus.fit(X_b,y_b)
clf_mult.fit(X_b,y_b)
scores_b = [cross_val_score(clf_ber,X_b,y_b).mean(),
            cross_val_score(clf_gaus,X_b,y_b).mean(), 
            cross_val_score(clf_mult,X_b,y_b).mean()]
scores_b

[0.62742040285899936, 0.9367492806089297, 0.89457904019307521]

Максимальное - 0.9367492806089297, GaussianNB

Выберите верные утверждения и запишите их номера через пробел (в порядке возрастания номера):

1) На вещественных признаках лучше всего сработал наивный байесовский классификатор с распределением Бернулли

2) На вещественных признаках лучше всего сработал наивный байесовский классификатор с мультиномиальным распределением

3) Мультиномиальное распределение лучше показало себя на выборке с целыми неотрицательными значениями признаков

4) На вещественных признаках лучше всего сработало нормальное распределение

3 4